angular/src/app/modules/timesheet/timesheet-detail/timesheet-detail.component.ts

```diff
@@ -418,6 +419,39 @@ export class TimesheetDetailComponent extends PagedListingComponentBase<Timeshee
     this.contextMenuPosition.y = e.clientY + 'px';
     this.menu.openMenu();
   }
+
+  SelectAllChanged(event){
+    if(event.checked){
+      this.TimesheetDetaiList = this.TimesheetDetaiList.map((item) => {
```

Comment:
tìm hiểu kỹ xem map trong js chạy ntn rồi tối ưu lại đoạn code này.
đừng gán lại this.TimesheetDetaiList